src/web-component/ay-accordion-header/index.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Ayogo Health Inc
+
+/**
+ * ay-accordion-header acts as a button for its first parent ay-accordion element thereby enabling the toggle functionality
+ *
+ * For example:
+ * ```
+ * <ay-accordion-header>
+ *  Button Name
+ * </ay-accordion-header>
+ * ```
+ *
+ * @name ay-accordion-header
+ *
+ */
+
+const accordionHeaderClickMap : WeakMap<HTMLElement, () => void> = new WeakMap();
+const accordionHeaderPressMap : WeakMap<HTMLElement, (event: KeyboardEvent) => void> = new WeakMap();
+
+export class AyAccordionHeader extends HTMLElement {
+
+  connectedCallback() {
+    const ayAccordionElem = this.closest('ay-accordion') as HTMLElement;
+
+    this.setAttribute('role', 'button');
+    this.setAttribute('tabIndex', '0');
+
+
+    const toggleOnClick = () => {
+      let toggleEvent = new Event('toggle');
+      console.log('dispatching click event from header');
+      ayAccordionElem.dispatchEvent(toggleEvent);
```

Comment:
1. It would be better to look up the closest ayAccordion element every time we dispatch an event, in case stuff has moved around in the DOM.

2. It's valid to just use `el.dispatchEvent(new Event('toggle'))`, although you might need to handle older browsers where you'd need to use `document.createEvent('toggle')` instead